lib/charms/vault_k8s/v0/vault_managers.py

```diff
@@ -0,0 +1,348 @@
+"""Library for managing Vault Charm features.
+
+This library encapsulates the business logic for managing the Vault service and
+its associated integrations within the context of our charms.
+"""
```

Comment:
Can we add a bit more content here. It would be great to have:
- The definition of a manager
- What belongs here, and what doesn't
- Any design principle as to what a manager should look like (Ex. with the `sync` approach)

Reply:
Let me know what you think: https://github.com/canonical/vault-k8s-operator/pull/534/commits/7454c11fd16a2d5951808dd205ff143f266cee2e

Reply:
I think it's great, thank you!